### DISCUSSION TITLE
How to handle longer CSV values

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#4352] [Feature Requests]
`CSV` is a Directus superset type that maps to different database types based on vendor. In some vendors this maps to `VARCHAR`, limiting how many characters/items it can save. Some users needs to save more values, but there is no control over the length of this.

### We could either...
* Move all to `TEXT` (instead of `VARCHAR`)
* Use the length field to determine which DB type to use
* Have both `CSV` (`VARCHAR`) and `BIGCSV` (`TEXT`) superset types

**Comment by rijkvanzanten** [comment]
> Move all to TEXT (instead of VARCHAR)

This sounds like the most straightforward way to me. Just like with a JSON blob, there's no real reason to limit the amount of characters allowed within a CSV value.

**Comment by philleepflorence** [comment]
Depending on if indexing, full-text search, et al, is required on the CSV column, alternating between `VARCHAR` and `TEXT` can make a difference - I usually start with `VARCHAR` and update to `TEXT` when needed.

It would be nice to allow the **Admin** to change the **Data Type** and **Length** as needed or:

- Have both `CSV` (`VARCHAR`) and `BIGCSV` (`TEXT`) superset types.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
